content/flare/consensus.md

```diff
@@ -287,3 +287,223 @@ Market forces cause quorum slices to overlap and this gives rise to the network-
+
+### Characteristics
+
+Modern fault-tolerant distributed systems are often built on multicast group communication protocols, which impose a partial order of messages and on many of them information is replicated on several (or all) processors in order to achieve faster access and protection against faults.
+Maintaining the consistency of replicated information is an important and problematic aspect of the implementations, and it can be simplified by imposing a total order, rather than a partial order, on messages.
+The total order on messages ensures that exactly the same updates are applied to the replicated data in the same order at all nodes.
+
+The Total Ordering algorithms are executed independently and concurrently by each process and each of them incrementally creates a total order, without requiring the transmission of any additional messages in the general case.
+They operate in the presence of asynchrony and communication faults, which may cause different processes to receive messages in different orders, and they tolerate both crash and Byzantine process fault.
+The system is asynchronous in the sense that no bound can be placed on the time required for a computation or for a communication of a message.
+A crash fault causes a process to stop processing and generating messages.
+A Byzantine fault allows an arbitrary behavior by the faulty process, including generating of messages intended to mislead the other processes.
+
+The total order ensures consistency by having the same updates applied to the replicated pieces of data in the same order at all sites.
+
+Each broadcast message has a header consisting of a process identifier, a message sequence number and payload content.
+A typical implementation might use a digital signature to substantiate the source of a message (process identifier) for each node.
+Systems capable of tolerating Byzantine faults must be able to confirm the source of a message and to detect messages that have the same header but different contents and/or acknowledgements within the message.
+Another option is to also embed a digest of the message content in its identifier, so that acknowledgements of messages with different contents but same sequence number will be regarded as acknowledgements of distinct messages.
+The acknowledgement and re-transmission mechanisms can ensure that all versions of a message are delivered to all destinations.
+Digital signatures and digests are computationally expensive, but do not require the additional rounds of message exchange of alternative strategies.
+
+The output of the algorithms is a total order of messages that is identical at all non-faulty processes in the system.
+Message `x1` from process `a` follows a message `x0` from process `b` if `x1` acknowledges `x0` or `x1` acknowledges some other message `y` that follows `x0`, where `y` is not the same as `x0` and `y` is not the same as `x1`.
```

Comment:
This sentence kind of lost me 😄 I had to read it twice to get it. Maybe this is the kind of situation where a small diagram would be clearer than a sentence? WDYT?